### DISCUSSION TITLE
IFC file problem with BIM2SIM BPS-EnergyPlus

### DISCUSSION THREAD

**Discussion by Xingjian-Wang** [BIM2SIM/bim2sim#484] [Q&A] [answered]
Hi there,
I attempted to import my own IFC file into BIM2SIM for BPS simulation and received a message as shown in the attached screenshot stating that 2029 related elements were found, but there are still 518 IfcBuildingElementProxy elements that are undefined. I checked the IFC export options in Revit, and each building component used in modeling has its own IFC class, as shown in the attached screenshot. I would like to know how to view these undefined building components so that I can select the appropriate decision. If it is not possible to view detailed information about these building components, can I select the "skip all" option to continue with the simulation?
![IfcBuildingElementProxy Screenshot](https://user-images.githubusercontent.com/72249299/219639421-e431122c-44a9-4b2e-b3ee-3e637debbc7d.png)
![IFC export option screenshot](https://user-images.githubusercontent.com/72249299/219639029-f6333dd1-73ce-4d50-9c79-79ccfd7cceb2.png)

I also noticed a specific task in EnergyPlus called IfcValidation, and I tried to use it to validate my IFC file. I wrote two lines of code to read my IFC file, and when I saved and attempted to run it, I did not see any error messages, but I also did not find a result file about validating the IFC file. I would like to ask if this method of validating an IFC file is incorrect and how to validate an IFC file in advance.The Python file is attached as well.
[ifc_validation_test.zip](https://github.com/BIM2SIM/bim2sim/files/10766196/ifc_validation_test.zip)

Thanks in advance

Best Regards,
Xingjian Wang

**Comment by veronikarichter** [comment, marked as answer]
As the Proxy issue is caused by the exported IFC, I would recommend analyzing the entities in your IFC before loading it into the bim2sim tool. You can use any BIM viewer you like. You could also try to use "skip all" if you ensure that no important information was exported as proxy. 

The EnergyPlus validation script was not implemented to be used as a standalone tool, so you need to set the input and output files according to your own needs. The implementation of the validation included in the BIM2SIM tool is very basic and no longer maintained, but feel free to extend it. 

We decided to develop a standalone tool for validation (as described [here](https://publications.ibpsa.org/conference/paper/?id=simbuild2022_C033)) that is not publicly available yet (but we plan to publish it in 2023).

**Reply by Xingjian-Wang** [reply]
Hi Veronika,

thank you very much for your response. I have made adjustments to my BIM model and reconfigured the IFC export options. After these changes, I did not see any prompts regarding the ifcBuildingElementProxy during the BIM2SIM simulation. The new issue is that after the simulation completed, I noticed in the log that BIM2SIM prompted the creation of a temporary folder named with "bim2sim" prefix, and the simulation results were generated in a folder named "export" located within that folder. However, the "export" folder is actually empty. I would like to know how to resolve this issue and what pre-settings I need to make.

Many thanks in advance

Regards,
Xingjian Wang

**Reply by veronikarichter** [reply]
If the export directory is empty, either the simulation failed or the project path was not set correctly (so the results are somewhere else than you expect).

**Reply by Xingjian-Wang** [reply]
Thanks for the response. I tried with almost 15 different ifc files(the files with same architecture model but different export settings and some of them I deleted the thermal zones ) I got only one successful simulation which I could find the idf file in the export folder, so it means the path is correct. But unfortunately that the only one can not run successfully in EnergyPlus, I got 2 severe errors but no further information about why it failed. I would like to ask you if there is a possibility of knowing why it failed to run by bim2sim. I checked the log file and it showed me no errors, is anywhere I can check?
Many thanks for considering my request.

**Reply by veronikarichter** [reply]
If the EnergyPlus simulation fails, you should always look at the EnergyPlus error reports and have a look at the EnergyPlus input files (IDF). Even if the BIM2SIM Tool is designed to generate EnergyPlus input files and should be able to run an EnergyPlus simulation, the user is asked to check the validity of the generated EnergyPlus input file before using the results. Small modifications may be needed. For the severe errors, you need to dive deeper into the EnergyPlus error files. If you are unfamiliar with the use of EnergyPlus, I highly recommend the EnergyPlus documentation and [UnmetHours](https://unmethours.com/questions/). 

For checking the validity of EnergyPlus input, I recommend using a combination of EnergyPlus Launcher for checking the general parameters and OpenStudio for visualization. 

Also, please make sure your IFC are in the IFC4 format and provide IfcRelSpaceBoundary (2ndLevel) information. We'll shortly add a validation task that raises errors, if these requirements are not fulfilled for the IFC file you're trying to simulate. (see #170).
I don't know though why you deleted the thermal zones from your IFC file.

**Reply by Xingjian-Wang** [reply]
Many for your prompt reply and tips.
I checked the error report in energyplus and the error should come from the window I used and I will fix that. I would also like to ask if it is possible to find the reason why bim2sim cannot generate idf files and if there is a possibility of knowing why it failed to run by bim2sim.
My ifc file is in IFC4 format and provides IfcRelSpaceBoundary (2ndLevel) information, there is no hint about CheckIfcBPS as mentioned in issues170.
The reason I removed the thermal zones and room names from some of the ifc files is that during my previous simulations with bim2sim I encountered garbled code as shown in the attached image, and for some reason some rooms were not recognized by bim2sim. In my project except for a server room to provide cooling and heating other spaces are only heating and set to the same temperature, my temporary idea is to let bim2sim first recognize the server room and then in idf can continue to edit the other areas.
![image](https://user-images.githubusercontent.com/72249299/221631331-14d851d1-f2ae-43d2-805b-4c3a49f7665c.png)
